src/Umbraco.Core/ContentApps/ContentEditorContentAppFactory.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Umbraco.Web.ContentApps
 {
-    internal class ContentEditorContentAppFactory : IContentAppFactory
+    public class ContentEditorContentAppFactory : IContentAppFactory
```

Comment:
Any reason these are all be made public, there's quite a few in this PR but not sure how that's related to this PR? Just wondering if we are ever going to re-review what should and shouldn't be public if we are just changing everything to public

Reply:
It is related because internals from core is not visible to infrastructure. IMO a good thing, because it makes it harder to test.

I think it is very important to take revisit the internal/public discussion. But whether we should use real `internal` is most likely the big question. The only reason to do it now is to avoid introducing breaking changes, right?

Reply:
Yes without 'real' internal makes things very difficult to deal with breaking changes in the future. There's a few 'challenges' internally that we already need to deal with in regards to breaking changes so if we even further restrict what we can change it's going to be very very difficult to change things into the future without a new major version or cutting through a lot of red-tape. 

On the note of 'real' vs non real internal things, i don't agree that non-real internal things works such as having an 'internal' namespace. This is what Microsoft tried with netcore and it failed which is why they've gone back to real internals since everyone was just using the internal namespaces and then complaining when they broke it. 

But anyways, we just need to make sure that we somehow save some time to go through all the APIs that we don't actually need to have as public before release else we may end up shooting ourselves in the foot.